### PR TITLE
Avoid crash for `XDG_CONFIG_HOME=/dev/null`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/4199.

## Test Plan

`XDG_CONFIG_HOME=/dev/null cargo run venv`

### CODE REVIEW

File: crates/uv-workspace/src/workspace.rs [hunk 1]
```diff
@@ -26,7 +26,17 @@ impl Workspace {
         debug!("Loading user configuration from: `{}`", file.display());
         match read_file(&file) {
             Ok(options) => Ok(Some(Self { options, root })),
-            Err(WorkspaceError::Io(err)) if err.kind() == std::io::ErrorKind::NotFound => Ok(None),
+            Err(WorkspaceError::Io(err)) if matches!(err.kind(), std::io::ErrorKind::NotFound) => {
```
Comment (ibraheemdev) [inline crates/uv-workspace/src/workspace.rs:29]: Why this change? Just curious.

**Review by ibraheemdev — APPROVED** [review]

File: crates/uv-workspace/src/workspace.rs [hunk 2]
```diff
@@ -26,7 +26,17 @@ impl Workspace {
...
+                Ok(None)
+            }
+            Err(_) if !dir.is_dir() => {
+                // Ex) `XDG_CONFIG_HOME=/dev/null`
+                debug!(
+                    "User configuration directory `{}` does not exist",
```
Comment (BurntSushi) [inline crates/uv-workspace/src/workspace.rs:35]: I might say, "... does not exist or is not a directory"

**Review by BurntSushi — APPROVED** [review]

File: crates/uv-workspace/src/workspace.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-workspace/src/workspace.rs:29]: Oops, mistake. I started by changing it to match against NotADirectory.
